solutions/swb-app/src/environmentTypeConfigRoutes.ts

```diff
@@ -61,37 +51,28 @@ export function setUpEnvTypeConfigRoutes(
   router.get(
     '/environmentTypes/:envTypeId/configurations',
     wrapAsync(async (req: Request, res: Response) => {
-      const { paginationToken, pageSize } = req.query;
-      if ((paginationToken && typeof paginationToken !== 'string') || (pageSize && Number(pageSize) <= 0)) {
-        res
-          .status(400)
-          .send('Invalid pagination token and/or page size. Please try again with valid inputs.');
-      } else {
-        const envTypeConfig = await environmentTypeConfigService.listEnvironmentTypeConfigs(
-          req.params.envTypeId,
-          pageSize ? Number(pageSize) : undefined,
-          paginationToken
-        );
-        res.send(envTypeConfig);
-      }
+      const listEnvTypeConfigRequest = validateAndParse<ListEnvironmentTypeConfigsRequest>(
+        ListEnvironmentTypeConfigsRequestParser,
+        { envTypeId: req.params.envTypeId, ...req.body }
+      );
+
+      const envTypeConfig = await environmentTypeConfigService.listEnvironmentTypeConfigs(
+        listEnvTypeConfigRequest
+      );
+      res.send(envTypeConfig);
     })
   );
 
   // Update envTypeConfig
-  router.put(
+  router.patch(
```

Comment:
Patch is supposed to do partial update, it seems to be more like put method to me

Reply:
its a partial, the update params are optional , so , not sending all props will filter on `update` method and ignore undefined values